### ISSUE TITLE
'Generating cache, one moment' runs indefinitely or at least for several minutes

### ISSUE THREAD

**Issue by LinuxOnTheDesktop** [linuxmint/mintinstall#315]
**What happened**

I ran MintInstalI, for the first time since updating to Mint 20.2, and it showed, 'Generating cache, one moment' for approximately five minutes, at which point I closed the program  (using the close button) and then restarted the program. Adter that restart, I had the same problem.

I see nothing that seems relevant in the syslog or dmesg.

**What I expect to happen**

If the generation of the cache will takes ages - even on a relatively fast PC - then I expect to be given a somewhat accurate indication of how long the process will take. Perhaps though the program has hung.

**My system**

MintInstall 8.0.8

OS: Linux Mint 20.2 x86_64 
Host: ThinkPad X1 Carbon 6th 
Kernel: 5.8.0-59-generic 
Packages: 3331 (dpkg) 
DE: Cinnamon 
CPU: Intel i7-8550U (8) @ 4.000GHz 
GPU: Intel UHD Graphics 620 
Memory: 2361MiB / 15902MiB 
Disk (/dev/nvme0n1p4): 15G / 40G (41%) 
Disk (/dev/nvme0n1p2): 172G / 405G (45%)

EDIT: the problem persisted after a reboot of the computer.

**Comment by mtwebster** [comment]
Can you run mintinstall from a terminal and paste any output?

**Comment by LinuxOnTheDesktop** [comment]
Of course. I should have thought of that. The output - from one computer; I tried two - is below.

Here is the thing though. On both computers, running mintinstall from the terminal worked: there was a delay of some ten seconds and then all was as it should be. Moreover, having running mintinstall from the terminal, running it via GUI worked (on both computers).

Here is the output (from one computer - the one that I found had the problem; I do not know whether the other one had the problem).

```
MintInstall: Detected system architecture: 'x86_64'
Installer: User pkgcache too old, skipping
Installer: User pkgcache too old, skipping
Installer: Generating new pkgcache
Installer: flatpak - metadata for remote 'flathub' has been updated. Comparing appstream timestamps...
Installer: flatpak - no new appstream data for remote 'flathub', skipping download
Installer: Processing Flatpaks for cache took 3372.469 ms
'NoneType' object has no attribute 'summary'
Installer: Processing APT packages for cache took 3138.628 ms
Full installer startup took 7208.285 ms
MintInstall: Deleting old screenshots
MintInstall: Downloaded new reviews
```
So I wonder what was going wrong before. More to the point perhaps: mintinstall should detect and respond to problems. (Or so it seems to me.)
